ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -293,6 +301,27 @@ private BlockImportResult importBlockAndState(
     return result;
   }
 
+  public void updateLatestValidFinalizedSlot(final ExecutePayloadResult payloadResult) {
+    onForkChoiceThread(
+            () -> {
+              if (payloadResult.getStatus() != ExecutionPayloadStatus.VALID) {
+                return;
+              }
+              UInt64 latestValidFinalizedSlotInStore =
+                  recentChainData.getStore().getLatestFinalized().getSlot();
+
+              if (recentChainData
+                  .getLatestValidFinalizedSlot()
+                  .map(value -> value.compareTo(latestValidFinalizedSlotInStore) != 0)
```

Comment:
This should definitely only update if the new value is greater than the old one.